subsys/bluetooth/host/classic/rfcomm.c

```diff
@@ -1507,16 +1510,30 @@ static int rfcomm_recv(struct bt_l2cap_chan *chan, struct net_buf *buf)
 
 	LOG_DBG("session %p dlci %x type %x", session, dlci, frame_type);
 
-	fcs_len = (frame_type == BT_RFCOMM_UIH) ? BT_RFCOMM_FCS_LEN_UIH :
-		   BT_RFCOMM_FCS_LEN_NON_UIH;
+	if (BT_RFCOMM_LEN_EXTENDED(hdr->length)) {
+		msg_len = BT_RFCOMM_GET_LEN_EXTENDED(hdr_ext->hdr.length, hdr_ext->second_length);
+		hdr_len = sizeof(*hdr_ext);
+	} else {
+		msg_len = BT_RFCOMM_GET_LEN(hdr->length);
+		hdr_len = sizeof(*hdr);
+	}
+
+	if (buf->len < (hdr_len + msg_len + 1)) {
+		LOG_ERR("Too small RFCOMM information (%d < %d)", buf->len, hdr_len + msg_len + 1);
+		return 0;
+	}
+
+	fcs_len = (frame_type == BT_RFCOMM_UIH)         ? BT_RFCOMM_FCS_LEN_UIH
+		  : BT_RFCOMM_LEN_EXTENDED(hdr->length) ? 1 + BT_RFCOMM_FCS_LEN_NON_UIH
+							: BT_RFCOMM_FCS_LEN_NON_UIH;
```

Comment:
The magic number `1` is being used a lot in this function, but it's not 100% clear what it is.

Suggest to define it somehow (either using a `#define` or define a variable for the value and use `sizeof` so that there's some context for the value.

Reply:
I have optimised the patch.